### ISSUE TITLE
Invalid results file when using allure report file in publish results test step

### ISSUE THREAD

**Issue by jwerkau** [Microsoft/vsts-tasks#4111]
My main goal is to be able to build a report that includes attaching a screenshot from a failed test case to a report which can be uploaded in the Publish Test Results step.  I have used the allure reports before and it is set up to easily attach screenshots to a failed test result in the report file.

However after I run my build in TFS the Publish Test results step returns "Invalid results file. Make sure the Test Result Format field in the task matches the result format of the file:" I know that the allure reports follow the xUnit format however they also add in some supplementary data as well.  Attached 
is the resulting xml file.

[testsuite.txt](https://github.com/Microsoft/vsts-tasks/files/952639/testsuite.txt)

In short my question is does should TFS be able to read this report file?  Am I missing a step to allow TFS to read the file?

**Comment by prawalagarwal** [comment]
Hi @jwerkau,

In the Publish Test Results task, we parse the results file to get to the relevant nodes to get test cases information. For that we strictly follow the XSD schema for Xunit (http://xunit.github.io/docs/format-xml-v2.html). The file you have attached does not conform to the Xunit schema and hence the error. 

Please let us know if you have any other questions.

**Comment by jwerkau** [comment]
Thanks @prawalagarwal.  Makes sense.  It is odd that they claim to use a standard xunit format but as you point out it is not following the XSD schema.  Rather than building the report and publishing through the publish results step I will install the Allure plugin in TFS and try generating the report that way in order to display the results.

**Comment by prawalagarwal** [comment]
Thanks @jwerkau. Closing the thread.
